cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2ClusterTest.java

```diff
@@ -1635,6 +1636,31 @@ public void testPodSetWithOAuthWithAccessToken() {
         });
     }
 
+    @ParallelTest
+    public void testPodSetWithServiceAccountOAuth() {
```

Comment:
Should this test also when it is used for the source clusters?

Reply:
I added the test by pattern-matching other OAuth tests, and they only test with target cluster. I guess the reasoning here was that we really test `KafkaClientAuthenticationOAuthBuilder` here, which is used both for the target and for the source cluster, thus if it works for one why would it fail for the other?

Reply:
The env vars work differently for source clusters as there might be many of them. That is why I think it needs to be tested as well. It is possible it was forgotten for the previous part, but you can probably add it at least for this feature.